### DISCUSSION TITLE
不留缺失的瑰宝！云之幻，请听我说说我的理由和感受。

### DISCUSSION THREAD

**Discussion by grandzzr** [Richasy/Bili.Uwp#1725] [想法]
云之幻，你好！

首先，我要感谢你为bilibili的用户提供了一个优秀的客户端，让我们能够更方便地观看视频，享受娱乐和学习。你的客户端不仅界面美观，功能强大，还兼容多种平台，真是一款神器！

但是，我想向你提出一个小小的建议，希望你能够考虑一下。那就是：能否在客户端中加入下载视频的功能呢？

我知道，这个功能可能会涉及到版权和流量的问题，也可能会增加你的开发和维护的难度。但是，我相信，这个功能对于很多用户来说，是非常有用和有意义的。

为什么呢？因为有些用户可能会遇到以下的情况：
- 有些用户可能会在没有网络或者网络不稳定的地方观看视频，比如地铁、飞机、山区等。如果能够提前下载好视频，就可以随时随地观看，不用担心卡顿或者断网。
- 有些用户可能会想要收藏一些自己喜欢的视频，以便于以后回味或者分享。如果能够下载视频到本地，就可以永久保存，不用担心视频被删除或者下架。
- 有些用户可能会想要对视频进行一些二次创作，比如剪辑、配音、字幕等。如果能够下载视频到本地，就可以更方便地进行编辑和处理。

总之，下载视频的功能可以让用户更自由地控制自己的观看体验，也可以让用户更深入地参与到bilibili的文化中去。

当然，我也理解你可能会有一些顾虑和困难。比如：你可能会担心这个功能会侵犯视频作者的版权和利益。但是，我想说的是，大部分用户都是尊重版权和原创的，他们下载视频并不是为了盗版或者转载，而是为了更好地欣赏和支持。而且，你也可以在客户端中设置一些限制和提示，比如只允许下载已经授权或者免费的视频，或者在下载前提醒用户遵守版权法规等。

下载视频的功能是一个非常有价值和有需求的功能，它可以让你的客户端更加完善和强大，也可以让你的用户更加满意和忠诚。我希望你能够认真考虑一下我的建议，如果你能够实现这个功能，我将不胜感激！

最后，我再次感谢你为bilibili的用户做出的贡献，你的客户端是我最喜欢的一个！我期待着你的回复和更新！

**Comment by Richasy** [comment]
抱歉回复晚了，你可以试试我新做的 [哔哩助理](https://github.com/Richasy/Bili.Copilot)，它提供下载功能，但是依赖BBDown，不过配置并不复杂，你应该能很快上手
